asdf/commands/tests/test_edit.py

```diff
@@ -0,0 +1,231 @@
+    YAML file with the same number of characters in the YAML section as was in
+    the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_smaller.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            if "foo" in l:
+                print("foo: 2", file=fd)  #  Change a value
+            else:
+                fd.write(l)
+
+    return oname
+
+
+def create_edit_larger(base_yaml):
+    """
+    The YAML from the base ASDF file will have a 'foo' value.  After this
+    line, add another line.  This will create an edited YAML file that will
+    have more characters than the YAML portion of the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_larger.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            fd.write(l)
+            if "foo" in l:
+                print("bar: 13", file=fd)  # Add a line
+
+    return oname
+
+
+def copy_base_asdf_equal(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with equal number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_equal.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+def copy_base_asdf_smaller(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with equal number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_smaller.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+def copy_base_asdf_larger(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with a larger number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_larger.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+@pytest.mark.parametrize("version", asdf.versioning.supported_versions)
+def test_edits(tmpdir,version):
+    #        Test:
+    # Create base ASDF file for testing
+    tmpdir = "/Users/kmacdonald/tmp"
+    asdf_base = create_base_asdf(tmpdir, version)
+
+    # Create base YAML file from base ASDF file
+    base, ext = os.path.splitext(asdf_base)
+    yaml_base = f"{base}.yaml"
+    # Run: asdftool edit -e -f {asdf_base} -o {yaml_base}
+    args = ["edit", "-e", "-f", f"{asdf_base}", "-o", f"{yaml_base}"]
+    main.main_from_args(args)
+
+    # Test smaller
```

Comment:
I would split each of these cases into separate parametrized tests.  That has the advantage of clearly identifying the failed case when something goes wrong.  As it stands, if the "smaller" case fails, we won't know if the other two would have succeeded since the test halts at the first error.

That structure would also save us from having to copy the "base" ASDF file, since we'd be only using it once per test anyway.

Reply:
The test has been refactored creating three separate tests.